### ISSUE TITLE
Data architecture

### ISSUE THREAD

**Issue by arthus-leroy** [Aceship/AN-EN-Tags#27]
I am a bit confused about the current architecture of the data.
We have json/gamedata. Should we consider it to be the language or the region ?
* If we consider it language, all the elements should be populated and translated (which is not the case, it's populated as it would be for region - the unreleased operators in the given region are not available)
* if we consider it region, all multilingual information should be in translation files tl-* (which is not the case as a part of the internationalization data is in gamedata.

In the current state, we can't provide the content of each server in each language (CN-related content would always be in Chinese or partially in English, heck I even seen some parts that assume English or interchangeably use lang and reg).

Imo, the best way to organize data is to have 1 global file for items, 1 for characters, ... which would hold all the meaningful info and have each internationalization data be kept in gamedata with any language-dependent info be replaced by an id (or equivalent). Also a list of existing items / chars / ... should be maintained for each server for easy manipulation (adding a new operator to a server, for exemple).

Does this seems reasonable ? It's a big change and it would be better to do it before changing any language-dependent page as it would change the way data is queried and used.
I can take care of it if needed (iff translation is provided of course).

**Comment by Aceship** [comment]
tbh i used 3 different data (probably more cause i didnt really tidying it up)

currently the usage is like this (but not all) :
- first is the main language for all the content (like cn region data for character detail)
- second is using the en language for the translation (for chara voiceline and story, skill and talent isnt using this one)
- then use community translated data (also for cn region, since no official english translation yet)

~mostly its used like that because im not used with internationalization~ 

much more problem is the way i automatically translate the story chara top part, especially when there's no translation yet, but i guess it could be used with other language with a few tweaks 

basically im using all the game data as both region and translation when needed

and tl- for missing one (except skill and talent and base , because official en translation have kinda inconsistent translation)

**Comment by arthus-leroy** [comment]
Wouldn't it be better to reorganize data to take care of both region and language ? Do we *care* about CN server, or is the region currently useless/unused/unusable ?

**Comment by Aceship** [comment]
well if possible to be used on other language too, but tbh most chara detail stuff gonna need tweaks , especially for multiple language (like changing how the data loads at the beginning)

~and personally too lazy to do it, mostly cause there's still feature in works like chibi and map stage~

**Comment by arthus-leroy** [comment]
Yeah, was wondering how much it would affect if I changed the data architecture and if we it was actually worth the shot. Will probably do it later after cleaning the pages and lat the foundations for such a transition :thinking:
